src/ASM/LR/ASMu3Dmx.C

```diff
@@ -352,9 +354,11 @@ bool ASMu3Dmx::integrate (Integrand& integrand,
       double   param[3] = { 0.0, 0.0, 0.0 };
       Vec4     X(param,time.t);
 
-      int iEl = el->getId();
+      int iEl = els[itgBasis-1]-1;
       fe.iel = MLGE[iEl];
 
+      el = lrspline->getElement(iEl);
```

Comment:
Since the `el` variable is declared const (in L344), is it fine to assign a new value to it here?  Thought the point of const'ing it was to prevent if from being redefined (or maybe there is a pointer issue that still allows it).

Consider changing L348 into
```
this->getElementsAt(threadBasis->getElement(groups[t][e])->midpoint(),els,elem_sizes);
```
instead (yes, it then becomes longer than 80 chars), and moving the type declaration of el into L360, such that the code is comparable with the equivalent 2D version.

Reply:
Yes. it's a pointer to a const LR::Element, not a const pointer to a const LR::Element. If it was
```
const LR::Element* const el
``` 
then it points to a const element AND i cannot re-assign the el variable.

in any case i've made the requested change.